### PR TITLE
Fix growroot when using software raid

### PR DESCRIPTION
**Pull request by jovial**
Using @markgoddard's suggestion from: https://github.com/stackhpc/stackhpc-kayobe-config/pull/770#issuecomment-1814402925

### CODE REVIEW

File: releasenotes/notes/fixes-growroot-for-software-raid-3852bdea5415a0be.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    Fixes a regression when using growroot.yml and software raid where the
```
Comment (Alex-Welsh) [inline releasenotes/notes/fixes-growroot-for-software-raid-3852bdea5415a0be.yaml]: nit: backticks
```suggestion
    Fixes a regression when using ``growroot.yml`` and software raid where the
```

File: etc/kayobe/ansible/growroot.yml [hunk 2]
```diff
@@ -77,7 +77,7 @@
           vars:
             pv: "{{ pvs.stdout | from_json }}"
             disk_tmp: "{{ pv.report[0].pv[0].pv_name[:-1] }}"
-            disk: "{{ disk_tmp[:-1] if disk_tmp[-1] == 'p' and disk_tmp[:4] == 'nvme' else disk_tmp }}"
+            disk: "{{ disk_tmp[:-1] if pv.report[0].pv[0].pv_name | regex_search('[a-z0-9]+[0-9]+p[0-9]+') else disk_tmp }}"
```
Comment (markgoddard) [inline etc/kayobe/ansible/growroot.yml:78]: I assume you tested it? :)

Comment (priteau) [reply etc/kayobe/ansible/growroot.yml:78]: I am trying it as I ran into the same issue moments ago, but it doesn't work for me…

Comment (priteau) [reply etc/kayobe/ansible/growroot.yml:78]: We end up using `/dev/md` because the last digit is removed in `disk_tmp`.

Comment (priteau) [reply etc/kayobe/ansible/growroot.yml:78]: Growing software RAID devices appears to require a more complex procedure anyway? https://documentation.suse.com/sles/12-SP5/html/SLES-all/cha-raid-resize.html

Comment (jovial) [reply etc/kayobe/ansible/growroot.yml:78]: Hmm, its been working for me. Were you trying this exact change?

Comment (jovial) [reply etc/kayobe/ansible/growroot.yml:78]: It kind of seems like you might have been doing the regex_search on disk_tmp rather than the full pv name

Comment (jovial) [reply etc/kayobe/ansible/growroot.yml:78]: > Growing software RAID devices appears to require a more complex procedure anyway? https://documentation.suse.com/sles/12-SP5/html/SLES-all/cha-raid-resize.html

I'm setting my md size to MAX when I create it:
```
{
  "logical_disks": [
    {
      "size_gb": "MAX",
      "raid_level": "1",
      "controller": "software",
      "is_root_volume": true,
      "physical_disks": [
        {"by_path": "/dev/disk/by-path/pci-0000:00:11.5-ata-5"},
        {"by_path": "/dev/disk/by-path/pci-0000:00:11.5-ata-6"}
      ]
    }
  ]
}
```
I just need to grow the partitions/lvm data-structures.

Comment (jovial) [reply etc/kayobe/ansible/growroot.yml:78]: This is my example:
```
[stack@cpt01 ~]$ lsblk                                                                
NAME                       MAJ:MIN  RM   SIZE RO TYPE  MOUNTPOINTS
sda                          8:0     0 223.6G  0 disk  
└─sda1                       8:1     0   223G  0 part  
  └─md126                    9:126   0 222.9G  0 raid1 
    ├─md126p1              259:4     0   500M  0 part  /boot/efi
    ├─md126p2              259:5     0     8M  0 part  
    ├─md126p3              259:6     0 222.3G  0 part  
    │ ├─rootvg-lv_root     253:0     0    50G  0 lvm   /
```
pv.report[0].pv[0].pv_name: /dev/md126p3
disk_tmp: /dev/md126p
disk: /dev/md126

Comment (jovial) [reply etc/kayobe/ansible/growroot.yml:78]: Possibly would be less fragile to use some of the ideas from here: https://unix.stackexchange.com/questions/718742/find-the-partition-and-block-device-for-a-filesystem-without-string-manipulati

**Comment by Alex-Welsh** [comment]
What's the state of this PR? I'm holding off running an overcloud host configure at IfA because `--check --diff` errors out and I think this is what's causing it.

**Comment by Alex-Welsh** [comment]
Is this change still required @jovial? https://github.com/stackhpc/stackhpc-kayobe-config/pull/947 has since merged

**Comment by mnasiadka** [comment]
> #947

It is required, we've hit that in SMS LAB yesterday on one sw-raid host, because we match for name only.

File: etc/kayobe/ansible/growroot.yml (hunk 2 above)
Comment (mnasiadka) [reply etc/kayobe/ansible/growroot.yml:78]: Ok, tested with:
- /dev/md127p3
- /dev/sdx3
- /dev/nvmen1p3

Works with all of these, can we just merge this and get over with it?

**Review by mnasiadka — DISMISSED** [review]

**Review by mnasiadka — APPROVED** [review]

**Pull request merged**